sdk/go/common/apitype/core.go

```diff
@@ -318,8 +318,10 @@ type ResourceV3 struct {
 	PendingReplacement bool `json:"pendingReplacement,omitempty" yaml:"pendingReplacement,omitempty"`
 	// AdditionalSecretOutputs is a list of outputs that were explicitly marked as secret when the resource was created.
 	AdditionalSecretOutputs []resource.PropertyKey `json:"additionalSecretOutputs,omitempty" yaml:"additionalSecretOutputs,omitempty"`
-	// Aliases is a list of previous URNs that this resource may have had in previous deployments.
-	Aliases []resource.URN `json:"aliases,omitempty" yaml:"aliases,omitempty"`
+	// AliasURNs is a list of previous URNs that this resource may have had in previous deployments.
+	AliasURNs []resource.URN `json:"aliases,omitempty" yaml:"aliases,omitempty"`
+	// Aliases is a list of structured Alias objects that this resource may have been associated with in previous deployments.
+	Aliases []resource.Alias `json:"structuredAliases,omitempty" yaml:"structuredAliases,omitempty"`
```

Comment:
Would this be a breaking change to a core type? Aliases used to be []resource.URN but now []resource.Alias?

Reply:
That's correct, however the JSON and YAML representations of them didn't change, just the internal name.  I wouldn't expect that to break anything, but please correct me if I'm wrong!

Reply:
Ah! Interesting! How's that you use custom serializes?

I found this:

https://github.com/pulumi/pulumi/pull/10819/files#diff-b913ac779219263c220553a986939580340f13b3dd9d354e09a7a15851e1d3beR8

```
type Alias struct {
	URN     URN
	Name    string
	Type    string
	Project string
	Stack   string
	Parent  URN
}
```

But no MarshalJSON()  hook. Perhaps that's fine since you're marshaling manually.

```
// In the future, we may add elements to the URN; it is more important that it is unique than it is
// human-typable.
type URN string
```

Another consideration here is that this change goes in the `sdk/*` code, which is supposed to have more stringent bw-compat discipline than `pkg` code. Specifically it's a minor breaking change to Go programs that  link against the older version of the SDK, even if the wire representation is identical. Someone updating the SDK needs to go edit their code a little bit.

That might be ok but we need to call it out in the changelog. And of course if there's a way to do it without the little breaking change that's even better.

Reply:
Oh interesting - do you mean that this could break if someone updated their pulumi binary, but not their Go SDK?

Reply:
More like if they were updating their Go SDK in the go.mod they'd need to make changes. Perhaps that's expected?